src/main/java/org/opensearch/security/dlic/rest/api/RolesApiAction.java

```diff
@@ -91,7 +91,7 @@ private ValidationResult<JsonNode> validateMaskedFields(final JsonNode content)
 
         private Pair<String, String> validateMaskedFieldSyntax(final JsonNode maskedFieldNode) {
             try {
-                new MaskedField(maskedFieldNode.asText(), SALT).isValid();
+                new MaskedField(maskedFieldNode.asText(), SALT, "").isValid();
```

Comment:
The algorithm should be checked during the validity check - lets plumb this through

Reply:
Thanks @peternied  I considered this and my findings were:

- This isn't plumbed through for the existing, configured salt in this instance either. 
- If the new configured default algorithm were invalid the node would not have started in the first place.

If needed, which I don't think it is, how can the setting be easily referenced here?

Reply:
@terryquigleysas `opensearchSettings` are passed as a constructor arg in this class.

Reply:
Once this is addressed, the rest looks good to me. It doesn't appear like the algo matters for the validation check, but I agree that we should pass the algo into this call.

One change we could consider making in the future is to track the supported algorithms in an Enum so that its not a generic String, but the list of supported algorithms will change depending on the version of the JDK used.

Reply:
> @terryquigleysas The settings can be extracted from the `securityApiDependencies`. These dependencies are passed to all API handlers from [here](https://github.com/opensearch-project/security/blob/main/src/main/java/org/opensearch/security/dlic/rest/api/SecurityRestApiActions.java#L69).
> 
> `securityApiDependencies.settings()` - https://github.com/opensearch-project/security/blob/main/src/main/java/org/opensearch/security/dlic/rest/api/SecurityApiDependencies.java

@cwperks Thanks. Code updated.